src/agreement/mod.rs

```diff
@@ -489,15 +493,20 @@ impl<NodeUid: Clone + Debug + Ord> Agreement<NodeUid> {
         sender_id: &NodeUid,
         msg: CommonCoinMessage,
     ) -> AgreementResult<FaultLog<NodeUid>> {
-        let mut fault_log = self.common_coin.handle_message(sender_id, msg)?;
+        let coin_step = self.common_coin.handle_message(sender_id, msg)?;
         self.extend_common_coin();
+        self.on_coin_step(coin_step)
+    }
 
-        if let Some(coin) = self.common_coin.next_output() {
+    fn on_coin_step(
+        &mut self,
+        coin_step: CommonCoinStep<NodeUid>,
+    ) -> AgreementResult<FaultLog<NodeUid>> {
+        let mut fault_log = coin_step.fault_log.clone();
```

Comment:
Maybe destructure then? We still the `output` value.

```rust
let CommonCoinStep { fault_log, output } = coin_step;
```

Reply:
@afck, what if the user wants to clone `FaultLog`? Wouldn't we violate the common traits convention if we remove `Clone` from a public type?

Reply:
Maybe, yes. Depends on whether it's more helpful to prevent accidental cloning or to allow cloning.
Not sure; I was just thinking about our own code, where we'd probably want the former.

Reply:
OK, I'll try to remove `Clone` from `FaultLog` then.